test/e2e/upgrade/manifestdelete/manifest-delete.go

```diff
@@ -0,0 +1,187 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	restclient "k8s.io/client-go/rest"
+	apiregistrationsclientset "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+// UpgradeTest runs verifies invariants regarding what alerts are allowed to fire
+// during the upgrade process.
+type UpgradeTest struct {
+	oc     *exutil.CLI
+	config *restclient.Config
+}
+
+type object struct {
+	kind      string
+	namespace string
+	name      string
+}
+
+type unstructured struct {
+	name      string
+	namespace string
+	group     string
+	version   string
+	resource  string
+}
+
+var (
+	// Update this array to contain objects which should have been deleted in this release. For example:
+	//	object{kind: "Namespace",
+	//		namespace: "",
+	//		name:      "openshift-foobar"},
+	//
+	deletes = []object{}
+
+	// Update this array to contain unstructured objects which should have been deleted in this release. For example:
+	//	unstructured{name: "dns",
+	//		namespace: "openshift-dns-operator",
+	//		group:     "monitoring.coreos.com",
+	//		version:   "v1",
+	//		resource:  "prometheusrules"},
+	//
+	unstructuredDeletes = []unstructured{}
+)
+
+func (obj object) uniqueName() string {
+	if len(obj.namespace) == 0 {
+		return obj.name
+	}
+	return obj.namespace + "/" + obj.name
+}
+
+func (obj object) String() string {
+	return fmt.Sprintf("%s \"%s\"", obj.kind, obj.uniqueName())
+}
+
+func (u unstructured) uniqueName() string {
+	if len(u.namespace) == 0 {
+		return u.name
+	}
+	return u.namespace + "/" + u.name
+}
+
+func (u unstructured) String() string {
+	return fmt.Sprintf("unstructured \"%s/%s, %s, %s\"", u.group, u.version, u.resource, u.uniqueName())
+}
+
+func (UpgradeTest) Name() string { return "check-for-deletes" }
+func (UpgradeTest) DisplayName() string {
+	return "[sig-arch] Verify object deletions after upgrade success"
+}
+
+// Setup creates parameters to query Prometheus
+func (t *UpgradeTest) Setup(f *framework.Framework) {
+	g.By("Setting up upgrade delete test")
+	oc := exutil.NewCLIWithFramework(f)
+	t.oc = oc
+	config, err := framework.LoadConfig()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	t.config = config
+	framework.Logf("Post-upgrade delete test setup complete")
+}
+
+// Test checks if alerts are firing at various points during upgrade.
+// An alert firing during an upgrade is a high severity bug - it either points to a real issue in
+// a dependency, or a failure of the component, and therefore must be fixed.
+func (t *UpgradeTest) Test(f *framework.Framework, done <-chan struct{}, upgrade upgrades.UpgradeType) {
+	if len(deletes) == 0 && len(unstructuredDeletes) == 0 {
+		framework.Logf("No object deletions in this release to verify")
+		return
+	}
+
+	// Block until upgrade is done
+	g.By("Waiting for upgrade to finish before checking for deletes")
```

Comment:
I'm a bit fuzzy in the origin/Ginkgo setup, but can we use `[sig-arch][Late] Verify...` above to get slotted in at the end, and then drop this block?  Not a big deal either way.